Modules/docker

akarzim opened the pull request:
Add some concise aliases for docker, docker-machine and docker-compose.

AFAIK, theses aliases will not shadow other system commands or prezto’s module aliases.

oohlaf committed:
Improve and simplify titling functions.

Multiplexer titles can be set independent of window and tab titles.
Add auto-title always option to have dynamic titling inside terminal
multiplexers. Setting auto-title to yes continues to ignore dynamic
titling inside terminal multiplexers.

This patch should not alter behavior under Apple terminals.

zmwangx committed:
utility module: fix/improve colordiff call in diff

The new call is compatible with colordiff v1.0.8 - v.1.0.15 (might be
compatible with even earlier versions, but not tested), while the
original one breaks down under v1.0.14 and v1.0.15. See
https://github.com/daveewart/colordiff/issues/22.

posva committed:
Added option to skip git aliases definition

Fixes #947

Eriner committed:
Move common promptpwd code to external function

jfilip committed:
Use less destructive git push --force-with-lease.

kireledan committed:
Added support for python venv display in paradox prompt

akarzim committed:
Add bundle clean alias

akarzim committed:
Doc (ruby): add doc on bundle clean alias

akarzim committed:
Feat (aliases): add aliases for git flow

malikoth committed:
Add "sa" alias to make it easy to search aliases

leoj3n committed:
Remove manb from completion filename

paulmelnikow committed:
Don't load pfunctions from vi swap files

jlax47 committed:
Change $TMPDIR for non-interactive shells

This commit changes the way $TMPDIR is set by using `mktemp` rather
than a fixed string.

mrgrauel committed:
update brewu method

Homebrew decided to not change the behaviour of `brew upgrade` so
`brew upgrade --all` is equivalent to `brew upgrade` without any other
arguments (so the `--all` is a no-op and can be removed).

paulmelnikow commented:
For the record: this is a repost of sorin-ionescu/prezto#1147

malikoth commented:
Exciting!

paulmelnikow commented on modules/docker/README.md:
```diff
@@ -0,0 +1,187 @@
+
+[1]: https://www.docker.com/
+[2]: https://github.com/zplug/zplug
+[3]: https://github.com/akarzim/zsh-docker-aliases/issues
```
This seems a bit confusing. We've started some discussions (see #38 and #33) about these blocks.

I'd like this to read something like this:

> ## Support
>
> If you're having problems, use the [Prezto issue tracker][].
>
> ## Acknowledgements
>
> This module is a copy of [akarzim/zsh-docker-aliases][] by [François Vantomme](https://github.com/akarzim) (MIT License).

[Prezto issue tracker]: https://github.com/zsh-users/prezto/issues
[akarzim/zsh-docker-aliases]: https://github.com/akarzim/zsh-docker-aliases

cc @johnpneumann @Cottser who have been part of those discussions

star-szr replied:
@paulmelnikow 👍 that makes sense to me, seems like a good way to acknowledge the origin of the code without signing them up to support it :)

akarzim replied:
Thanks for the feedback. I agree with this. Changes are done ;)

paulmelnikow commented:
I don't want to set a precedent for adding lots of new modules. However Docker is widely used, so I'd like to include it in Prezto.

I'm interested to hear what other people think.

malikoth commented:
I'm very much in favor of including this.  Docker seems to have swept the world, and it would be nice to have this module maintained to make it easier and faster to work with Docker.

belak committed:
Allow setting ZPREZTODIR to make prezto easier to integrate with

paulmelnikow commented:
Looks good to me.

belak committed:
Set ZPREZTODIR to ${0:h}

maximbaz commented:
> I don't want to set a precedent for adding lots of new modules.

@paulmelnikow why not? #52 might take quite a long time, why prevent adding new modules until it's done? When #52 is ready, all the external plugins will likely be purged anyway.

jrolfs committed:
Add nodenv support to Node.js module

- Implementation from https://github.com/sorin-ionescu/prezto/pull/1001
- Documentation from https://github.com/sorin-ionescu/prezto/pull/1178

Plastix committed:
Update pure prompt to v1.5.2
This fixes a security issue with malicious branch names

paulmelnikow commented:
Just a heads up that reviewing and merging are on pause because Sorin is open to adding maintainers to the original (#44).

johnpneumann committed:
Adds additional support for PS prompt context expansion

ezintz committed:
Use python3 for http-serve alias if available (#1193)

mateuszlewko committed:
Ensure code blocks are formatted properly for GitHub (#1271)

eukaryote committed:
Add support for skipping VIRTUALENVWRAPPER init in modules/python (#598)

* Support skip-virtualenvwrapper-init zstyle in python module to avoid auto-sourcing of virtualenvwrapper.sh
* Don't overwrite the WORKON_HOME env var if the user has already set it

paulmelnikow commented:
@belak has started merging changes at the original. I sincerely hope this is merged! It looks like the original PR now needs to be rebased, as it seems to be tracking the same branch as this one.

Think the simplest thing would be to make a new branch, and a new PR at the original.

d125q committed:
Avoid unsetting nonexistent variables (#1075)

Eriner committed:
Prefer $(COMMAND) syntax over backticks (#1012)

zmwangx committed:
Remove extra use of grep in git-hub-shorten-url (#904)

astropanic committed:
Add notice about potential shadowing of the gb command (#1124)

belak committed:
Update zsh-autosuggestions submodule location

Thanks to @naibaf0. Closes #1096.

belak committed:
Update zsh-users submodules to latest stable versions

belak committed:
Update external prompt modules

paulmelnikow committed:
Bump license years and tidy up readme (#1273)

tough-griff committed:
Use patterns from ruby module for initializing nodenv

tough-griff committed:
Add support for iTerm2 v3 in tab function

tough-griff committed:
Fix path in node module

johnpneumann committed:
Merge pull request #1272 from sorin-ionescu/issue/1246-ps-context-config

Adds additional support for PS prompt context expansion

nstcactus committed:
FIXED: syntax error in osx module tab function

belak committed:
Only try to run python-info if it exists for the paradox prompt

belak committed:
Ensure extendedglob is set in promptpwd

Solisol committed:
Update fasd README with basic installation information

belak commented:
There's a bit of discussion that I'd like to see in the new PR before it gets merged... things like what aliases are worth including and how they should be organized. In short, I feel like the prezto git module aliases get a bunch of things right and I'd like to see a similar structure with the docker aliases. Anyway, discussion at the new PR, so this can probably be closed.

ezintz committed:
Allow syntax highlighting pattern styles (#1192)

catharsis committed:
completion: Cap max-errors at 7 to avoid hanging (#953)

7 is pretty arbitrarily chosen, but seems like a reasonable tradeoff, at
least the completion no longer shows symptoms of exponential
time-growth when trying to complete something completely wrong.

This fixes #946.

belak committed:
Add note about working around preexisting runcoms

Closes #955

belak committed:
Don't use a deprecated argument as a python example

Thanks to @jeffwidman for pointing this out

Closes #1141

akarzim committed:
Feat (docker): add docker aliases (#1147)

Thanks to @malikoth and @saoula for their input.

belak committed:
Fix indentation in README

belak committed:
Update is-callable to also check builtins

Fixes #1037

paulmelnikow commented:
Merged via sorin-ionescu/prezto#1147.
